modules/snooker/src/main/ts/ephox/snooker/api/Generators.ts

```diff
@@ -166,12 +169,53 @@ const merging = function (generators: Generators) {
       // Remove any width calculations because they are no longer relevant.
       Css.remove(raw, 'width');
       Css.remove(cell, 'width');
+
+      scope.each((attribute) => Attribute.set(raw, 'scope', attribute));
+
       return raw;
     };
   };
 
+  const merge = (cells: SugarElement[]) => {
+    const getScopeProperty = () => {
+      const attributeOpts = Arr.map(cells, (cell) => {
+        const attributeOpt = Attribute.getOpt(cell, 'scope');
+
+        return attributeOpt.map((attribute) => attribute.substr(0, 3)); // Attribute can be col, colgroup, row, and rowgroup.
+        // As col and colgroup are to be treated as if they are the same, lob off everything after the first three characters and there is no difference.
+      });
+
+      const stringAttributes = Optionals.cat(attributeOpts);
+
+      if (stringAttributes.length === 0) {
+        return Optional.none<string>();
+      } else {
+        const baseScope = stringAttributes[0];
+        const scopes = [ 'row', 'col' ];
+
+        const isMixed = Arr.exists(stringAttributes, (attribute) => {
+          return attribute !== baseScope && Arr.contains(scopes, attribute);
+        });
+
+        return isMixed ? Optional.none<string>() : Optional.from(baseScope);
+      }
+    };
+
+    Css.remove(cells[0], 'width');
+
+    const scope = getScopeProperty();
+
+    scope.fold(
```

Comment:
```suggestion
    getScopeProperty().fold(
```
Again the extra variable is never used elsewhere, so this is overly verbose.